RawSpeed/Cr2Decoder.cpp

```diff
@@ -190,6 +191,104 @@ void Cr2Decoder::decodeMetaDataInternal(CameraMetaData *meta) {
   if (mRootIFD->hasEntryRecursive(ISOSPEEDRATINGS))
     iso = mRootIFD->getEntryRecursive(ISOSPEEDRATINGS)->getInt();
 
+  // Fetch the white balance
+  if (mRootIFD->hasEntryRecursive(CANONCOLORDATA)) {
+    TiffEntry *color_data = mRootIFD->getEntryRecursive(CANONCOLORDATA);
+
+    // this entry is a big table, and different cameras store used WB in
+    // different parts, so find the offset
+    int offset = 0;
+    switch (color_data->count)
+    {
+      case 582: /* Canon EOS 20D, Canon EOS 350D */
+        offset += 50;
+      break;
+      case 653: /* Canon EOS-1D Mk II, Canon 1Ds Mk II */
+        offset += 68;
+      break;
+      case 674: /* Canon EOS-1D Mk III */
+      case 692: /* Canon EOS 40D */
+      case 702: /* Canon EOS-1Ds Mk III */
+      case 796: /* Canon EOS-1D Mk II N, Canon EOS 5D, Canon EOS 30D, Canon EOS 400D */
+      case 1227: /* Canon EOS 450D, Canon EOS 1000D */
+      case 1250: /* Canon EOS 5D Mk II, Canon EOS 50D */
+      case 1273: /* Canon EOS 600D */
+      case 1312: /* Canon EOS 5D Mk III, Canon EOS 700D */
+      case 1313: /* Canon EOS 70D */
+      case 1316: /* Canon EOS-1D X */
+      case 1337: /* Canon EOS-1D Mk IV, Canon EOS 7D */
+      case 1338: /* Canon EOS 550D */
+      case 1346: /* Canon EOS 60D, Canon EOS 1100D */
+        offset += 126;
+      break;
+      case 5120:
+        /* Canon PowerShot G10, Canon PowerShot G11, Canon PowerShot G12,
+         * Canon PowerShot G1X, Canon PowerShot S120, Canon PowerShot SX1 IS */
+        offset += 142;
+      break;
+      default:
+        std::cerr << "CR2 Decoder: CanonColorData has unsupported count of values: %d" << color_data->count << std::endl;
```

Comment:
Please don't print to stderr. Use writeLog(DEBUG_PRIO_INFO, "CR2 Decoder: CanonColorData has unsupported count of values: %d", color_data->count);